R/addin.R

```diff
@@ -69,29 +72,83 @@ confl_create_post_from_Rmd <- function(Rmd_file = NULL, interactive = NULL,
     env = globalenv()
   )
 
+  # set confl setting
   front_matter <- rmarkdown::yaml_front_matter(Rmd_file, "UTF-8")
 
+  front_matter_confl <- front_matter$confl_setting
+
+  confl_setting <- c()
+
+  # set confl setting: base
   if (!is.null(title)) {
-    front_matter$title <- title
+    confl_setting$title <- title
+  } else {
+    confl_setting$title <- front_matter$title
+  }
+
+  if (!is.null(args$tags)) {
+    confl_setting$tags <- args$tags
+  } else {
+    confl_setting <- front_matter_confl$tags
+  }
+
+  if (!is.null(args$space_key)) {
+    confl_setting$space_key <- args$space_key
+  } else {
+    confl_setting$space_key <- front_matter_confl$space_key
   }
 
+  if (!is.null(args$parent_id)) {
+    confl_setting$parent_id <- args$parent_id
+  } else {
+    confl_setting$parent_id <- front_matter_confl$parent_id
+  }
+
+  # set confl_setting for console
+  if (!interactive) {
+    if (!is.null(args$type)) {
+      confl_setting$type <- args$type
+    } else {
+      confl_setting$type <- front_matter_confl$type
+    }
+
+    if (!is.null(args$update)) {
+      confl_setting$update <- args$update
+    } else {
+      confl_setting$update <- front_matter_confl$update
+    }
+
+    if (!is.null(args$use_original_size)) {
+      confl_setting$use_original_size <- args$use_original_size
+    } else {
+      confl_setting$use_original_size <- front_matter_confl$use_original_size
+    }
+  }
```

Comment:
I think you can remove `args <- list(...)` above and simplify here as

``` r
confl_setting <- purrr::update_list(front_matter_confl, ...)
```

Reply:
Or, should we get rid of `...`?

``` r
confl_create_post_from_Rmd <- function(Rmd_file = NULL, interactive = NULL,
                                       title = NULL, tags = NULL, space_key = NULL, parent_id = NULL, update = FALSE, use_original_size = FALSE ..., params) {

(...snip...)

  if (interactive) {
    confl_addin_upload(
      md_file = md_file,
      title = title %||% confl_setting$title,
      tags = tags %||% confl_setting$tags,
      space_key = space_key %||% confl_setting$space_key,
      parent_id = parent_id %||% confl_setting$parent_id
    )
```

Reply:
I have never known about the function `purrr::list_update()` before.
However, this time I will use `purrr::list_modify()` instead of `purrr::list_update()`.